### DISCUSSION TITLE
What is the best way to convert a float to a string?

### DISCUSSION THREAD

**Discussion by marklacas** [tabemann/zeptoforth#34] [Q&A] [answered]
I've spent quite a bit to time trying to sort this out with no success so far.

22,37059 to a string and count for displaying on a ssd1306

**Comment by tabemann** [comment, marked as answer]
Use `format-fixed` ( c-addr f -- c-addr bytes ) or `format-fixed-truncate` ( c-addr f places -- c-addr bytes ), if you want to truncate the value to a certain number of places to the right of the decimal, well, comma. Note that the buffer starting at *c-addr* must have enough bytes available for the entire fixed-point value; I usually use 32 bytes just because it is a nice round number and has more than enough bytes to store a entire S31.32 fixed-point value (but note that it cannot store, say, an entire binary fixed-point number, and these words can format values in any base up to base 16).
